src/app/shared/form/builder/ds-dynamic-form-ui/models/affiliations/department/department-affiliation-select.component.ts

```diff
@@ -0,0 +1,161 @@
+   * 
+   * @param newValue: The new list of options to display.
+   * @param clearData: A flag to clear current value.
+   */
+  triggerDepartmentUpdate(newValue: AffiliationData[], clearData: boolean): void {
+    if (clearData) this.emptySelect();
+
+    if (!isEmpty(newValue)){
+      this.selectMode = true
+      this.affiliations  = newValue;
+      this.searchResult = this.affiliations;
+    } else {
+      // Since we have no options, disable the select mode.
+      this.selectMode = false;
+      this.affiliations = [];
+      this.searchResult = this.affiliations;
+    }
+  }
+
+  /**
+   * Emits a change event and set the current value with the given value.
+   * @param event The value to emit.
+   */
+  onSelect(event: AffiliationData) {
+    this.group.markAsDirty();
+    this.selectOption(event);
+  }
+
+  /**
+   * Handles the user input on the field. It emits a change event and sets the current value with the given value.
+   * @param event The value emitted due to user changes.
+   */
+  public onInput(event): void {
+    // If user input is empty, reset the select field.
+    if (isEmpty(event.target.value)) {
+      // Empty model value
+      this.emptySelect();
+      // Reset search results
+      this.searchResult = this.affiliations;
+      return;
+    }
+    let newValue = new FormFieldMetadataValueObject(event.target.value);
+    this.setCurrentValue(newValue);
+    this.dispatchUpdate(newValue);
+    this.updateSearch();
+  }
+
+  /**
+   * Updates the search result based on the user input.
+   */
+  private updateSearch(): void {
+    this.currentValue.subscribe((value) => {
+      if (value?.value) {
+        this.searchResult = this.affiliations.filter(
+          (affiliation: AffiliationData) => this.generateDisplayFormat(affiliation).toLowerCase().includes(value.value.toLowerCase())
+        );
+      } else {
+        this.searchResult = this.affiliations;
+      }
+    });
+  }
+
+  /**
+   * Selects the option chosen by the user and sets the current value.
+   * @param option: The selected option.
+   */
+  protected selectOption(option: AffiliationData) {
+    let newValue = new FormFieldMetadataValueObject(option.name, null, null, option.UUID, option.name);
+    this.setCurrentValue(newValue);
+    this.dispatchUpdate(newValue);
+    // Reset search results
+    this.searchResult = this.affiliations;
+  }
+
+  /**
+   * Empties the select field by resetting its current value.
+   */
+  protected emptySelect() {
+    this.setCurrentValue(undefined);
+    this.dispatchUpdate(undefined);
+  }
+
+  // Used to render option display value in the select field.
+  public affiliationRendering (affiliation: AffiliationData): string {
+    let spacing = '';
+    if (affiliation.index){
+        for (let i = 0; i < affiliation.index; i++) {
+          spacing += '\u00A0\u00A0\u00A0\u00A0';
+        }
+    }
+    return spacing + this.generateDisplayFormat(affiliation);
+  }
```

Comment:
It sounds like only templating method, I think it's better to place it into html template file or (better ?) using a pipe
```
{{ affiliation | renderDepthOption: affiliation.index }}
```
or "renderPaddedOption".
pipe arguments could be "depth" (required), "separator" (default = '\u00A0')